### ISSUE TITLE
[Bug] Cannot use output parameter due to it being hardcoded by task

### ISSUE THREAD

**Issue by quasarea** [GitTools/actions#282]
Moved from https://github.com/GitTools/GitVersion/issues/2473

**Describe the bug**
Running gitversion with /output file parameter gets overwritten. Actually it gets overwritten twice with two different parameters - json and buildserver.

```
C:\hostedtoolcache\windows\GitVersion.Tool\5.5.1\x64\dotnet-gitversion.exe D:/a/1/s /output json /output buildserver /config /output file
```

## Expected Behavior
Being able to specify intended output, perhaps as parameter

## Actual Behavior
buildserver output type gets precedence over both json and file outputs

## Possible Fix
Expose output as parameter

## Steps to Reproduce
Create task that will save GitVersion.json file using /output file parameter

## Context
I'm not able to create json artifact to reuse it in other jobs

**Comment by twilly86** [comment]
+1 also looking for a way to take json output and save to a file.

**Comment by chris-codeflow** [comment]
@quasarea.  I looked at the **GitVersion** tool's source code and it supports setting all three output options: _json_ (output to the console), _buildserver_ (output to build server variables) and _file_ (output to file in JSON format)

In order to output to a file, you ALSO need to specify the file path of the file with the `/outputfile <path>` parameter.  For example `/output file /outputfile semver.json`.

This is not a bug with the execute task.

**Comment by chris-codeflow** [comment]
@twilly86.  Please see my comment on how to output the **GitVersion** variables to a file in JSON format above.  Please also note that **GitVersion** only writes the output variables to a file in JSON format, no other formats are supported (at the time of writing).

**Comment by quasarea** [comment]
@chris-codeflow you are right that there are three options, unfortunately they are exclusive and only one is used. It prioritizes `buildserver` over `file` . So if you set both `/output buildserver` and `/output file` the buildserver will be used. 

The `/output buildserver` is set by task and there is no option to turn it off - therefore adding `/output file` is useless at the moment.

Additionally if you don't use `/outputfile`, the output will be saved as GitVersion.json.

@twilly86 here is my current workaround:
```
      - task: gitversion/setup@0
        inputs:
          versionSpec: '5.6.3'

      - pwsh: |
          @{
            Major = '$(GitVersion.Major)';
            Minor = '$(GitVersion.Minor)';
            Patch = '$(GitVersion.Patch)';
            PreReleaseTag = '$(GitVersion.PreReleaseTag)';
            PreReleaseTagWithDash = '$(GitVersion.PreReleaseTagWithDash)';
            PreReleaseLabel = '$(GitVersion.PreReleaseLabel)';
            PreReleaseNumber = '$(GitVersion.PreReleaseNumber)';
            WeightedPreReleaseNumber = '$(GitVersion.WeightedPreReleaseNumber)';
            BuildMetaData = '$(GitVersion.BuildMetaData)';
            BuildMetaDataPadded = '$(GitVersion.BuildMetaDataPadded)';
            FullBuildMetaData = '$(GitVersion.FullBuildMetaData)';
            MajorMinorPatch = '$(GitVersion.MajorMinorPatch)';
            SemVer = '$(GitVersion.SemVer)';
            LegacySemVer = '$(GitVersion.LegacySemVer)';
            LegacySemVerPadded = '$(GitVersion.LegacySemVerPadded)';
            AssemblySemVer = '$(GitVersion.AssemblySemVer)';
            AssemblySemFileVer = '$(GitVersion.AssemblySemFileVer)';
            FullSemVer = '$(GitVersion.FullSemVer)';
            InformationalVersion = '$(GitVersion.InformationalVersion)';
            BranchName = '$(GitVersion.BranchName)';
            EscapedBranchName = '$(GitVersion.EscapedBranchName)';
            Sha = '$(GitVersion.Sha)';
            ShortSha = '$(GitVersion.ShortSha)';
            NuGetVersionV2 = '$(GitVersion.NuGetVersionV2)';
            NuGetVersion = '$(GitVersion.NuGetVersion)';
            NuGetPreReleaseTagV2 = '$(GitVersion.NuGetPreReleaseTagV2)';
            NuGetPreReleaseTag = '$(GitVersion.NuGetPreReleaseTag)';
            VersionSourceSha = '$(GitVersion.VersionSourceSha)';
            CommitsSinceVersionSource = '$(GitVersion.CommitsSinceVersionSource)';
            CommitsSinceVersionSourcePadded = '$(GitVersion.CommitsSinceVersionSourcePadded)';
            UncommittedChanges = '$(GitVersion.UncommittedChanges)';
            CommitDate = '$(GitVersion.CommitDate)';
          } | ConvertTo-Json | Set-Content 'GitVersion.json'
        displayName: 'Generate GitVersion.json'

      - publish: '$(Build.SourcesDirectory)\GitVersion.json'
        displayName: 'Artifact: GitVersion'
        artifact: 'GitVersion'
```

and then reuse where needed
```
      - task: DownloadPipelineArtifact@2
        inputs:
          buildType: 'current'
          artifactName: GitVersion
          targetpath: $(Build.SourcesDirectory)

      - pwsh: |
          $GitVersion = ConvertFrom-Json -AsHashtable ( Get-Content -Raw -Path 'GitVersion.json' )

          Write-Host "InformationalVersion: $($GitVersion.InformationalVersion)"
          Write-Host "SemVer: $($GitVersion.SemVer)"
          Write-Host "Sha: $($GitVersion.Sha)"
          Write-Host "Build.BuildNumber: $(Build.BuildNumber)"

```

**Comment by chris-codeflow** [comment]
Hello @quasarea.  Thanks for your response to my comment.

I just checked the **GitVersion** tool's source code again and my understanding is that the output argument is NOT mutually exclusive and supports all three output types.  The argument parser treats the **Output** argument as a set and (the basis of my previous comment) the OutputGenerator will individually check and output if **buildserver**, **file** and **json** are in the set.

In fact, the hard-coded use of `/output json` and `/output buildserver` already prove that the `/output` argument is not mutually exclusive.

The `/output file` output type is special insofar as it also requires the `outputFile` argument to be supplied.

**Comment by chris-codeflow** [comment]
Hello @quasarea.  I just ran a GitHub actions test to verify the above.  It all worked as expected (it created output variables in JSON format, it created outputs for the GitHub build agent and it output the version variables to a specified file, which I proved by displaying the contents of the file in a separate step.

Snippet from action:

```
- name: Output version to file
  uses: gittools/actions/gitversion/execute@v0.9.8
  with:
    additionalArguments: '/output file /outputfile ./SemVer.json'

- name: Display version from file
  run: |
    cat ./SemVer.json
```

See [GitHub workflow file](https://github.com/chris-codeflow/gittools-actions-examples/blob/main/.github/workflows/execute.yml) for full details.
